staging/src/k8s.io/apiextensions-apiserver/test/integration/ratcheting_test.go

```diff
@@ -325,6 +327,51 @@ func (p patchMyCRDV1Beta1Schema) Description() string {
 	return p.description
 }
 
+// Verifies the ratcheting metrics exist, and the count metric has the given value
+type verifyMetrics struct {
+	description string
+	count       int
+}
+
+func (v verifyMetrics) Do(ctx *ratchetingTestContext) error {
```

Comment:
An integration test is pretty heavy handed, it's easier to write a unit test for this: https://github.com/kubernetes/kubernetes/blob/6330b2722559c49e76daf347240ba569023257e4/staging/src/k8s.io/apiserver/pkg/endpoints/filters/metrics_test.go#L36